### PR TITLE
New Article on provisioning Ubuntu-based EC2 instances with CDK

### PR DESCRIPTION
**Pull request by lmammino**
➡️ Post preview [here](https://github.com/lmammino/loige.co/blob/086d5364d81888d38fb112bb0356681420e29b33/content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md)

### CODE REVIEW

File: content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+
+## Ubuntu AMIs and where to find them
+
+When provisioning EC2 instances on AWS you have to provide a virtual machine image (an _AMI_ in AWS lingo). You can build and manage your own custom private images or use one of the many public images for different Linux distributions.
+
+These two approaches have their pros and cons:
+
+  - If you use a public image, you will need to provision all the needed software and configuration after the machine starts by using an init script. This is an easy and quick way to experiment and iterate over different versions of your virtual machines, but it will mean that the machine bootstrap might be slow, depending on how much stuff you need to do to prepare the environment.
+  - If you build your own custom images you'll need to have a pipeline (or some other reproducible approach) to bootstrap a base image, install all your software and create a new image from there. This path adds a bit more complexity but then your images will have everything built in and the bootstrap time will probably be much faster.
```
Comment (eoinsha) [inline content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md:123]: Maybe a node to [EC2 Image Builder](https://aws.amazon.com/image-builder/) is worthwhile here? Not a big deal.

File: content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+
+So, back to our main question: _how do we find a base image with Ubuntu installed_?
+
+There's an entire website dedicated to finding public AMIs for just Ubuntu: [Ubuntu Amazon EC2 AMI Locator](https://cloud-images.ubuntu.com/locator/ec2/).
+
+[![Ubuntu Amazon EC2 AMI Locator screenshot](./ubuntu-amazon-ec2-ami-finder-ui.png)](https://cloud-images.ubuntu.com/locator/ec2/)
+
+Every image is built for a specific AWS region, processor architecture and instance type. Every image is identified by a unique ID called _AMI ID_.
+
+When provisioning a new EC2 instance you need to specify the wantend _AMI ID_ and make sure to select the correct one for the region you are deploying your istance to.
```
Comment (eoinsha) [inline content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md:137]: `s/wantend/wanted`?

File: content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+    "Value": "ami-0298c9e0d2c86b0ed",
+    "Version": 221,
+    "LastModifiedDate": "2021-07-21T14:31:50.717000+01:00",
+    "ARN": "arn:aws:ssm:eu-west-1::parameter/aws/service/canonical/ubuntu/server/focal/stable/current/amd64/hvm/ebs-gp2/ami-id",
+    "DataType": "aws:ec2:image"
+  }
+}
+```
+
+As the name of the SSM parameter suggests, these SSM parameters are managed directly by Canonical (the company behind Ubuntu). You can also see that there is a specific path structure to identify the AMI ID.
```
Comment (eoinsha) [inline content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md:197]: I did not know this was possible 🤯 Cool!

**Review by eoinsha — APPROVED** [review]
Really nice! I learned a lot and found it interesting all the way through 💯

File: content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md (hunk 1 above)
Comment (lmammino) [reply content/posts/2021-08-04_provision-ubuntu-ec2-with-cdk/index.md:123]: I didn't know this one! Thanks 😊

**Pull request merged**